paddle/fluid/platform/mkldnn_reuse.h

```diff
@@ -39,7 +39,8 @@ class MKLDNNHandler {
     std::stringstream ss;
     ss << tid;
     key_ = key_common_ + "-t:" + ss.str();
-    if (platform::get_cur_thread_id() == -1) {
+    if (platform::get_cur_mkldnn_session_id() !=
```

Comment:
in original PR, -1 is used to indicate a special mode for cache clearing, and only this mode works in the case that not uses thread id in key. But I re-think it, -1 just means nothing but cache clearing, so if user sets thread id, it will isolate primitives with first key naturally, not need insert thread id into key. That's the initiative for this code change.